api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -600,6 +612,12 @@ private static String sanitizeString(CharSequence value, long now, int today) {
       } else if (TIMESTAMPTZ.matcher(value).matches()) {
         Literal<Long> ts = Literal.of(value).to(Types.TimestampType.withZone());
         return sanitizeTimestamp(ts.value(), now);
+      } else if (TIMESTAMPNS.matcher(value).matches()) {
```

Comment:
Is this code path used? The `TIMESTAMP` and `TIMESTAMPTZ` patterns match `\d{1,9}` and are checked first. I think those will likely match anything that these would match before checking the nanosecond patterns.

Reply:
Oops, reordered, and confirmed they actually are hit now.